### ISSUE TITLE
New version 2.0.3 seems not support python 3.9 and 3.10, only support 3.11+

### ISSUE THREAD

**Issue by LinkunGao** [beda-software/fhir-py#129]
```sh
from fhirpy import AsyncFHIRClient
  File "C:\Users\AppData\Local\anaconda3\envs\test\lib\site-packages\fhirpy\__init__.py", line 1, in <module>
    from .lib import AsyncFHIRClient, SyncFHIRClient
  File "C:\Users\AppData\Local\anaconda3\envs\test\lib\site-packages\fhirpy\lib.py", line 4, in <module>
    from fhirpy.base.client import TClient
  File "C:\Users\AppData\Local\anaconda3\envs\test\lib\site-packages\fhirpy\base\__init__.py", line 1, in <module>
    from .lib_async import AsyncClient, AsyncReference, AsyncResource, AsyncSearchSet
  File "C:\Users\AppData\Local\anaconda3\envs\test\lib\site-packages\fhirpy\base\lib_async.py", line 19, in <module>
    from fhirpy.base.searchset import AbstractSearchSet
  File "C:\Users\AppData\Local\anaconda3\envs\test\lib\site-packages\fhirpy\base\searchset.py", line 5, in <module>
    from typing import Generic, Self, Union
ImportError: cannot import name 'Self' from 'typing'
```

**Comment by ruscoder** [comment]
Hi, @LinkunGao, thanks for noticing that. It's fixed in 2.0.4

**Comment by LinkunGao** [comment]
Hi @ruscoder, seems there are still some bugs for fhirpy 2.0.4 running on Python 3.9, 3.10, and 3.11 now.

- The error is ModuleNotFoundError: No module named 'typing_extensions'
```sh
Traceback (most recent call last):
  File "C:\Users\Desktop\test\main.py", line 2, in <module>
    from fhirpy import AsyncFHIRClient
  File "C:\Users\AppData\Local\anaconda3\envs\test\lib\site-packages\fhirpy\__init__.py", line 1, in <module>
    from .lib import AsyncFHIRClient, SyncFHIRClient
  File "C:\Users\AppData\Local\anaconda3\envs\test\lib\site-packages\fhirpy\lib.py", line 4, in <module>
    from fhirpy.base.client import TClient
  File "C:\Users\AppData\Local\anaconda3\envs\test\lib\site-packages\fhirpy\base\__init__.py", line 1, in <module>
    from .lib_async import AsyncClient, AsyncReference, AsyncResource, AsyncSearchSet
  File "C:\Users\AppData\Local\anaconda3\envs\test\lib\site-packages\fhirpy\base\lib_async.py", line 19, in <module>
    from fhirpy.base.searchset import AbstractSearchSet
  File "C:\Users\AppData\Local\anaconda3\envs\test\lib\site-packages\fhirpy\base\searchset.py", line 8, in <module>
    from typing_extensions import Self
ModuleNotFoundError: No module named 'typing_extensions'
```

**Comment by ruscoder** [comment]
@LinkunGao Indeed, our CI/CD pipeline runs tests on different Python versions but without installation as the package that's why it was not handled. I've added typing-extensions as a required dependency. The fix will be available in 2.0.5
